python/ray/tune/tests/test_trial_runner_2.py

```diff
@@ -315,7 +315,8 @@ def testPauseThenResume(self):
         runner.add_trial(Trial("__fake", **kwargs))
         trials = runner.get_trials()
 
-        runner.step()
+        runner.step()  # Start trial
+        runner.step()  # Process result
```

Comment:
btw, why do we need to process results here?

Reply:
we want there to be a result associated with the checkpoint, otherwise we can't tell which checkpoint (persistent or memory) to return since its tied on training iteration (-1 for no result).